File: ts/src/platform.ts
/**
 * Platform dispatcher — routes send/receive calls to the correct transport module.
 */

import { sendMessage as sendWebexMessage, sendCard as sendWebexCard } from './webex.js';
import { sendSlackMessage, sendSlackCard } from './slack.js';
import { sendDiscordMessage, sendDiscordCard } from './discord.js';
import { sendIRCMessage, sendIRCCard } from './irc.js';

export async function platformSendMessage(
  platform: string,
  token: string,
  target: string,
  text: string,
  fetchFn?: typeof fetch,
): Promise<Record<string, unknown>> {
  if (platform === 'webex') {
    return sendWebexMessage(token, target, text, fetchFn);
  }
  if (platform === 'slack') {
    return sendSlackMessage(token, target, text, fetchFn);
  }
  if (platform === 'discord') {
    return sendDiscordMessage(token, target, text, fetchFn);
  }
  if (platform === 'irc') {
    return sendIRCMessage(token, target, text);
  }
  throw new Error(`Unsupported platform: ${platform}`);
}

export async function platformSendCard(
  platform: string,
  token: string,
  target: string,
  text: string,
  card: unknown,
  fetchFn?: typeof fetch,
): Promise<Record<string, unknown>> {
  if (platform === 'webex') {
    return sendWebexCard(token, target, text, card, fetchFn);
  }
  if (platform === 'slack') {
    return sendSlackCard(token, target, text, card, fetchFn);
  }
  if (platform === 'discord') {
    return sendDiscordCard(token, target, text, card, fetchFn);
  }
  if (platform === 'irc') {
    return sendIRCCard(token, target, text, card);
  }
  throw new Error(`Unsupported platform: ${platform}`);
}

File: python/src/wgrok/platform.py
"""Platform dispatcher — routes send/receive calls to the correct transport module."""

from __future__ import annotations

from . import discord, irc, slack, webex


async def platform_send_message(
    platform: str, token: str, target: str, text: str, session=None, room_id: str = "",
) -> dict:
    """Send a text message via the specified platform."""
    if platform == "webex":
        return await webex.send_message(token, target, text, session, room_id=room_id)
    if platform == "slack":
        return await slack.send_message(token, target, text, session)
    if platform == "discord":
        return await discord.send_message(token, target, text, session)
    if platform == "irc":
        return await irc.send_message(token, target, text, session)
    raise ValueError(f"Unsupported platform: {platform}")


async def platform_send_card(
    platform: str, token: str, target: str, text: str, card: dict, session=None, room_id: str = "",
) -> dict:
    """Send a message with card/rich content via the specified platform."""
    if platform == "webex":
        return await webex.send_card(token, target, text, card, session, room_id=room_id)
    if platform == "slack":
        return await slack.send_card(token, target, text, card, session)
    if platform == "discord":
        return await discord.send_card(token, target, text, card, session)
    if platform == "irc":
        return await irc.send_card(token, target, text, card, session)
    raise ValueError(f"Unsupported platform: {platform}")
